pages/real_time.py

```diff
@@ -0,0 +1,303 @@
+    Output('pause-button-realtime', 'disabled'),
+    Output('reset-button-realtime', 'disabled'),
+    Output('interval-realtime', 'disabled'),
+    Output('start-button-realtime', 'n_clicks'),
+    Output('pause-button-realtime', 'n_clicks'),
+    Output('reset-button-realtime', 'n_clicks'),
+
+    Output('asset-realtime', 'value'),
+    Output('input-container-realtime', 'children'),
+    Output('add-input-realtime', 'n_clicks'),
+
+    Input('start-button-realtime', 'n_clicks'),
+    Input('pause-button-realtime', 'n_clicks'),
+    Input('reset-button-realtime', 'n_clicks'),
+    Input('add-input-realtime', 'n_clicks'),
+
+    State('start-button-realtime', 'disabled'),
+    State('pause-button-realtime', 'disabled'),
+    State('reset-button-realtime', 'disabled'),
+    State('interval-realtime', 'disabled'),
+
+    State('asset-realtime', 'value'),
+    State('input-container-realtime', 'children'))
+def update_state(start_clicks, pause_clicks, reset_clicks, add_input_clicks,
+                 start_disabled, pause_disabled, reset_disabled, interval_disabled,
+                 asset, children):
+    global processing_job_name
+    # Args parsing
+    if add_input_clicks > 0:
+        name_input = dcc.Input(
+            id={'type': 'dynamic-input', 'index': add_input_clicks},
+            type='text',
+            placeholder='arg name'
+        )
+        value_input = dcc.Input(
+            id={'type': 'dynamic-input', 'index': add_input_clicks},
+            # type='number',
+            placeholder='value'
+        )
+        children.append(name_input)
+        children.append(value_input)
+        add_input_clicks = 0
+
+    if start_clicks > 0:
+        inputs = [child['props']['value'] for child in children]
+
+        for _ in inputs:
+            option = transformation(_)
+            process.append(option)
+
+    if asset:
+        process.append('-a')
+        process.append(asset)
+
+    # Handle start button click
+    if start_clicks > 0:
+        state['running'] = True
+        state['paused'] = False
+        start_disabled = True
+        pause_disabled = False
+        reset_disabled = False
+        interval_disabled = False
+        start_clicks = 0
+
+        # start processing_job
+        # processing_job_name = star_job()
+        # print(processing_job_name)
+
+        # Start data_generator for Windows
+        subprocess.Popen(process, shell=True)
+
+        # Start data_generator for Amazon linux
+        # subprocess.Popen(process)
+        print('start')
+
+    # Handle pause button click
+    if pause_clicks > 0 and state['running']:
+        state['paused'] = not state['paused']
+        pause_disabled = True if state['paused'] else False
+        start_disabled = False
+        interval_disabled = True
+        pause_clicks = 0
+
+        time.sleep(1)
+        # print(processing_job_name)
+        # stop_job(processing_job_name)
+        # Terminate data_generator
+        file = open('pid.txt', 'r')
```

Comment:
Always use context manager when reading files ( at least if you do not use some high-level method from a third party)